LX.StaffScheduler.BLL/Services/Common/CityService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using LX.StaffScheduler.BLL.DependencyInjection;
+using LX.StaffScheduler.BLL.DTO;
+using LX.StaffScheduler.BLL.Services.Interfaces;
+using LX.StaffScheduler.DAL.Interfaces;
+using LX.StaffScheduler.DAL.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LX.StaffScheduler.BLL.Services.Common
+{
+    public class CityService: ICityService
+    {
+        private readonly ICityRepository repository;
+
+
+        public CityService(ICityRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        public void Add(CityDTO entity)
+        {
+            repository.Add(entity.FromDTO());
+        }
+
+        public async Task<IEnumerable<CityDTO>> GetAll()
+        {
+            var cities = await Task.Run(() => repository.GetAll());
```

Comment:
not necessarily to use Task.Run cause a repository should have async version of methods and your call should be like
await _repository.GetAllAsync().